### PR TITLE
Update keymap.c

### PR DESCRIPTION
**Pull request by nshanpetrosyan**
Additional functionality added to layers.

## Description

Added Media Controls, and sub-legend support based on KBDfans XDA keycap set to **Layer 1**
Added Mac specific functionality to **layer 5** (brightness control, launchpad, mission control)
Changed Layout for RGB controls on **layer 2**
Additional bug fixes.

**LAYER REFERENCES BELOW**
image
image
image
image
image
image

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Fixes issue where toggling RGBs off made the FN indicator lights to stay on until restart.

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c [hunk 1]
```diff
@@ -8,38 +8,38 @@
...
-		KC_LCTL,  KC_LALT,  KC_LGUI,                KC_SPC,                KC_RALT,  MO(1)  , KC_LEFT,         KC_DOWN,           KC_RIGHT),
+		KC_LCTL,  KC_LALT,  KC_LGUI,                KC_SPC,                KC_RGUI,  MO(1)  , KC_LEFT,         KC_DOWN,           KC_RGHT),
 	[_LAYER1] = LAYOUT( /* FN */
 		TO(3),    KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_DEL , \
-		KC_TRNS,  KC_TRNS,  KC_UP,    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_CALC,  KC_TRNS,  KC_INS,   KC_TRNS,  KC_PSCR,  KC_SLCK,  KC_PAUS,  RESET  , \
-		KC_TRNS,  KC_LEFT,  KC_DOWN,  KC_RIGHT, KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_HOME,  KC_PGUP,            KC_TRNS, \
-		KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_END,   KC_PGDOWN, KC_VOLU,            KC_MUTE, \
-		KC_TRNS,  KC_TRNS,  KC_TRNS,                      TO(4),                        KC_TRNS,  KC_TRNS,  KC_MPRV,  KC_VOLD,            KC_MNXT),
+		KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    KC_NO,    LGUI(LSFT(KC_5)),   KC_SLCK,  KC_PAUS,  KC_NO,    KC_NO,   RESET,\
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: You can use `XXXXXXX` and `_______` (7 characters) in place of `KC_NO` and `KC_TRNS`, respectively. These make keymaps considerably easier to read :)

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c [hunk 2]
```diff
@@ -8,38 +8,38 @@
...
+		KC_NO,    KC_P0,    KC_PDOT,                      KC_ENT,                       KC_P0,    KC_PDOT,  KC_NO,    KC_NO,              KC_NO),
+	[_LAYER4] = LAYOUT( /* WIN */
 		KC_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,            KC_MINS,  KC_EQL,  KC_BSPC, \
 		KC_TAB,   KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,            KC_LBRC,  KC_RBRC, KC_BSLASH, \
 		CTL_T(KC_CAPS),  KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,         KC_QUOT,           KC_ENT, \
 		KC_LSFT,  KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   RSFT_T(KC_SLSH), KC_UP,             LT(2, KC_DEL), \
 		KC_LCTL,  KC_LGUI,  KC_LALT,                KC_SPC,                KC_RALT,  MO(5)  , KC_LEFT,         KC_DOWN,           KC_RIGHT),
 	[_LAYER5] = LAYOUT( /* FN */
-		TO(3),    KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_DEL , \
+		KC_BRMD,  KC_BRMU, LCTL(KC_UP), LSFT(KC_F12),     KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,   KC_F12,   KC_DEL , \
```
Comment (fauxpark) [inline keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: I recommend using `KC_BRIU` and `KC_BRID` here, as they will work on all OSes, whereas `KC_BRMU` and `KC_BRMD` are just aliases for Pause and Scroll Lock.

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c [hunk 3]
```diff
@@ -8,38 +8,38 @@
 #define _LAYER6 6
 #define _LAYER7 7
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
-	[_LAYER0] = LAYOUT( /* Base */
-		KC_GESC,  KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,            KC_MINS,  KC_EQL,  KC_BSPC, \
-		KC_TAB,   KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,            KC_LBRC,  KC_RBRC, KC_BSLASH, \
+	[_LAYER0] = LAYOUT( /* MAC */
+		KC_GESC,  KC_1, 	KC_2, 	  KC_3,  KC_4,  KC_5,  KC_6,	KC_7,  KC_8,	KC_9,	  KC_0,			   KC_MINS,  KC_EQL,  KC_BSPC, \
```
[the remark this answers is not in the document]
Comment (nshanpetrosyan) [reply keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: Just following Merlin's syntax, I was not sure if it is necessary. I'll remove them, thanks!

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c (hunk 1 above)
Comment (nshanpetrosyan) [reply keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: Oh, I did not know that! Thanks!

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (fauxpark) [reply keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: Yeah, it's an artifact of copy+paste, unfortunately (and was in the templates until recently). One day I'll get em all 😄

**Comment by nshanpetrosyan** [comment]
Seems like I'm missing a key, will update! thanks!

File: keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c (hunk 2 above)
Comment (nshanpetrosyan) [reply keyboards/dztech/dz60rgb/keymaps/mekanist/keymap.c]: Done, thank you :)

**Pull request merged**